src/handleLocalStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { EventData } from "./types/main.js";
+
+const currentEventDataKey: string = "currentEventData";
+const savedEventsKey: string = "allSavedEvents";
+
+const storeDataInLocalStorage = (key: string, data: EventData | EventData[]) => {
```

Comment:
Do you really want to store single item in the local storage? Also what is a mapping between the key and the data which is persisted?

I assume this API should look like

```
storeDataInLocalStorage('allSavedEvents', event) // if I pass 'allSavedEvents' as the first argument, second argument must be array of Events
```

Reply:
```
const LOCAL_STORAGE_KEY = {
   ALL_EVENTS = 'all-events',
} as const;

type LocalStorageKey = ObjectValues<typeof LOCAL_STORAGE_KEY>;

type LocalStorageData<T extends LocalStorageKey> = {
  [LOCAL_STORAGE_KEY.ALL_EVENTS]: { events: EventsData[] }
}[T] 


type StoreDataInLocalStorage<T extends LocalStorageKey> = (key: T, value: LocalStorageData[T]) => {
   ///
}
```